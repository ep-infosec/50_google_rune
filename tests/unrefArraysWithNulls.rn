//  Copyright 2021 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

class Foo8:u8(self, name) {
  self.name = name

  final(self) {
    println "Destroying ", self.name
  }
}

class Foo16:u16(self, name) {
  self.name = name

  final(self) {
    println "Destroying ", self.name
  }
}

class Foo32:u32(self, name) {
  self.name = name

  final(self) {
    println "Destroying ", self.name
  }
}

class Foo64:u64(self, name) {
  self.name = name

  final(self) {
    println "Destroying ", self.name
  }
}

func arrayWithNullEntries() {
  l1 = [Foo8("Alice8"), null(Foo8), Foo8("Bob8")]
  l2 = [Foo16("Alice16"), null(Foo16), Foo16("Bob16")]
  l3 = [Foo32("Alice32"), null(Foo32), Foo32("Bob32")]
  l4 = [Foo64("Alice64"), null(Foo64), Foo64("Bob64")]
}

arrayWithNullEntries()
