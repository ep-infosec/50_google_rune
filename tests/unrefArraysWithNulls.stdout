Destroying Alice64
Destroying Bob64
Destroying Alice32
Destroying Bob32
Destroying Alice16
Destroying Bob16
Destroying Alice8
Destroying Bob8
